homeassistant/components/modbus/climate.py

```diff
@@ -67,7 +72,8 @@
         vol.Optional(CONF_MAX_TEMP, default=35): cv.positive_int,
         vol.Optional(CONF_MIN_TEMP, default=5): cv.positive_int,
         vol.Optional(CONF_STEP, default=0.5): vol.Coerce(float),
-        vol.Optional(CONF_UNIT, default="C"): cv.string,
+        vol.Optional(CONF_STRUCTURE, default=DEFAULT_STRUCTURE_PREFIX): cv.string,
```

Comment:
I think we've already talked about this for another modbus platform:
We no longer allow new integrations to use a platform config section or old integrations to do changes to the platform config section. See https://github.com/home-assistant/architecture/blob/master/adr/0007-integration-config-yaml-structure.md.

Please refactor this integration to have the config schema in the base integration module and the config yaml section under the top level integration domain key.

Reply:
@MartinHjelmare Sure, I can do that. But this is not a new integration. If I change that, we'll break backwards compatibility. Please let me know if I should proceed.

Reply:
OK, I read the document. Basically, if I want to add a new config entry, the whole config section needs to be reworked.

@MartinHjelmare in PR #33642 you suggested to adapt the config entry in a way to not break compatibility. However, in this case, the compatibility will be broken. Perhaps it is time to go through the whole Modbus integration and move the configuration into a single place to meet your new coding guidelines once for all.

This change would involve changes inside main Modbus section (especially moving hub configuration into a separate subentry).

As an alternative, we can be more cautious, and change only what's needed for Climate and Cover integration. @janiversen, @MartinHjelmare what's your suggestion on this?

For now, I will prepare changes so that they'll break only Climate config.

Reply:
We should follow the cover platform example.

Reply:
Thjs is existing, so we should be very careful not to break any existing configurations.

I would really prefer to not mame breaking changes in climate, but wait a bit (as discussed earlier) and then change the whole of modbus.

Reply:
@janiversen the Home Assistant guidelines affect also existing integrations. According to the document, I can't add a new config entry under platform, which is deprecated now.

This change affects only the Climate entity. Perhaps I could check if the code is used elsewhere first, and if not, I think we should get this merged.

Reply:
Until the config structure is changed, no config option changes are allowed in this platform.

Reply:
OK, so do we keep this PR open until we do a bigger refactor of the whole Modbus?

Reply:
that is not fair against you, let us get it merged and deal with any side effects later

Reply:
Actually, I'm OK with that. The other changes are more important for me. This one is different - I ported what was already implemented in Modbus sensor into the Climate. It is handy feature to have, but it can wait.

However, if we decide to keep this open, somebody should ideally put a flag here, so that this PR won't be closed for being stale.

Let me know what you think it's best, and we'll go from there...

Reply:
If we're not planning to merge this soon, we should close the PR. It can be reopened later when we want to merge it.

Reply:
@janiversen what do you suggest? I am OK with both scenarios - postponing the change, or making necessary changes in the config to meet the requirements.

If we postpone, at least we should talk about the timeframe when we're going to introduce the necessary breaking changes to support new way of configuration.

Reply:
@MartinHjelmare I reworked the `Climate` configuration to be under the `Modbus` component (the same way as used for the `Cover`, and currently `Light` and `Fan`).

Please add the `breaking-change` label. Documentation was already updated.

Thanks!